api/src/main/java/com/microsoft/gctoolkit/jvm/PreUnifiedJavaVirtualMachine.java

```diff
@@ -20,14 +20,26 @@ public class PreUnifiedJavaVirtualMachine extends AbstractJavaVirtualMachine {
 
     private static final Logger LOGGER = Logger.getLogger(PreUnifiedJavaVirtualMachine.class.getName());
 
+    // Safepoint details are in a different log and would be loaded separately pre-unified.
     @Override
-    public boolean accepts(GCLogFile logFile) {
-        return ! logFile.isUnified();
+    public boolean accepts(DataSource logFile) {
```

Comment:
I think you can lift this into AbstractJavaVirtualMachine and get rid of both PreUnified and UnifiedJavaVirtualMachine.